server/cluster/cluster.go

```diff
@@ -504,6 +504,7 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		)
 		saveKV, saveCache, isNew = true, true, true
 	} else {
+		region.SetTimestamp(origin.GetTimestamp())
```

Comment:
region should be read-only after create from heartbeat, please see the detail comments about `RegionInfo`.

Reply:
I'll modify this.